### ISSUE TITLE
Disable Sliding Once Touch/Click Has Started

### ISSUE THREAD

**Issue by coofercat** [ismail9k/vue3-carousel#390]
**Describe the bug**
If using an interactive component, such as a slider, in a Carousel, interacting with the component causes the Carousel to move (as well as the component).

For example, if you have a left-right slider, you can slide it from side to side, but the carousel will move at the same time. The same is true if you use a click/touch start and end event in the component to unset the `mouseDrag` and `touchDrag` carousel props. However, if you (somehow) unset the `mouseDrag` and `touchDrag` props before you click on the slider, then it works as it should.

In summary then, it seems that once a click/touch has started, the `mouseDrag`/`touchDrag` props are ignored.

**To Reproduce**
Put a slider or other interactive component into a carousel. I'm personally using a [Colour Picker](https://github.com/stijlbreuk-dev/vue-color-picker-wheel?tab=readme-ov-file), although I've had to modify it to get the click start/end events to be able to set the Carousel props.

**Expected behavior**
If the `mouseDrag` and/or `touchDrag` props are set to false, then all carousel movement should stop, even if it's in progress already.

**Comment by Tofandel** [comment]
For your use case you should just stop propagation of the mousemove event from the components you want the user to interact with. The lib can't really make too many assumptions here

**Comment by coofercat** [comment]
Thanks for the pointers, I've been trying to achieve what you describe. However, it doesn't appear to be working. I put `event.stopPropagation()` in my slider, and can see that it does not propagate to the `div` that surrounds it, yet the Carousel slider still slides.

On further looking into it, I see that Javascript events occur in the Capture phase (events start in the root of the DOM and go "up" to the top most element), or the Bubbling phase (events go back down the DOM from the top most element to the root).

`stopPropagation()` stops events going "down" the DOM stack towards the root, so in theory is just what's required here. My slider 'consumes' the touch/mouse events and the Carousel never sees them. However, the Carousel seems to have event handlers in the Capture phase, so these are executed *before* the handlers in my slider, and so my slider can't prevent them from running.

Here's an example of an `addEventListener()` being called with an explicit `true` (ie. `{capture: true}` in it: https://github.com/ismail9k/vue3-carousel/blob/62015e12ec6d1695e0335b8d429b05ec68e53f2f/src/components/Carousel.ts#L201

**Comment by Tofandel** [comment]
I would suggest submitting a PR removing the capture flag from those events, they really have no business being there

I don't think the author of this lib understood how they worked because there is the same issue to prevent the click event at the end of dragging #396

**Comment by coofercat** [comment]
I'll look to get a TS dev environment together to try to work on that. (I'm not really a JS dev, I'm just doing some app work because there's no one else).

This feels like it might be disruptive to some of the more obscure use-cases. Probably worth dealing with that to get this straightened out though.

**Comment by coofercat** [comment]
I spent a couple of hours trying to get this all to work. Probably all my lack of understanding/experience/knowledge, but I couldn't even get the project dependencies to resolve properly, and after I'd messed about to get that "working" I got some Vue render-time errors related to refs which I couldn't understand much less diagnose. However, PR #398 is now raised.
